### PR TITLE
Add Trino 443 release notes

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Assemble the release notes for the upcoming Trino 443 release.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 15 Mar 2024

* #21077 ✅ rn ✅ docs
* #21094 ✅ rn ✅ docs
* #21095 ✅ rn ✅ docs
* #21090 ✅ rn ✅ docs
* #21091 ✅ rn ✅ docs

## 16 Mar 2024

* #20492 ✅ rn ✅ docs
* #21097 ✅ rn ✅ docs
* #21084 ✅ rn ✅ docs
* #21044 ✅ rn ✅ docs
* #21083 ✅ rn ✅ docs
* #21096 ✅ rn ✅ docs
* #21076 ✅ rn ✅ docs 
* #21114 ✅ rn ✅ docs
* #20489 ✅ rn ✅ docs
* #21103 ✅ rn ✅ docs
* #21108 ✅ rn ✅ docs
* #21113 ✅ rn ✅ docs
* #21110 ✅ rn ✅ docs
* #21115 ✅ rn ✅ docs

## 17 Mar 2024

- #21119 ✅ rn ✅ docs
- #21105 ✅ rn ✅ docs
- #21034 ✅ rn ✅ docs
- #20281 ✅ rn ✅ docs
- #21120 ✅ rn ✅ docs
- #21117 ✅ rn ✅ docs
- #21123 ✅ rn ✅ docs
- #21122 ✅ rn ✅ docs
- #21071 ✅ rn ✅ docs

## 18 Mar 2024

- #21124 ✅ rn ✅ docs
- #21133 ✅ rn ✅ docs
- #21107 ✅ rn ✅ docs
- #21116 ✅ rn ✅ docs
- #21098 ✅ rn ✅ docs
- #21101 ✅ rn ✅ docs
- #21137 ✅ rn ✅ docs
- #20960 ✅ rn ✅ docs
- #21075 ✅ rn ✅ docs
- #21139 ✅ rn ✅ docs
- #21129 ✅ rn ✅ docs
- #21128 ✅ rn ✅ docs
- #21142 ✅ rn ✅ docs
- #21144 ✅ rn ✅ docs
- #21146 ✅ rn ✅ docs
- #20892 ✅ rn ✅ docs
- #21150 ✅ rn ✅ docs

## 19 Mar 2024

- #21136 ✅ rn ✅ docs
- #21158 ✅ rn ✅ docs
- #21155 ✅ rn ✅ docs
- #21143 ✅ rn ✅ docs
- #21160 ✅ rn ✅ docs
- #21141 ✅ rn ✅ docs
- #21166 ✅ rn ✅ docs

## 20 Mar 2024

- #21169 ✅ rn ✅ docs
- #21170 ✅ rn ✅ docs
- #21171 ✅ rn ✅ docs
- #21172 ✅ rn ✅ docs
- #21168 ✅ rn ✅ docs
- #21125 ✅ rn ✅ docs
- #21181 ✅ rn ✅ docs

## 21 Mar 2024

- #21126 ✅ rn ✅ docs
- #21179 ✅ rn ✅ docs
- #21185 ✅ rn ✅ docs
- #21186 ✅ rn ✅ docs
- #21188 ✅ rn ✅ docs
- #21145 ✅ rn ✅ docs
- #20814 ✅ rn ✅ docs
- #21190 ✅ rn ✅ docs
- #21187 ✅ rn ✅ docs
- #21192 ✅ rn ✅ docs
- #21184 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-443.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+* Ensure consistent formatting when casting from double or real to varchar. ({issue}`21136`)
+* Prevent query failure when partial aggregation over long decimals is pushed
+  below joins. ({issue}`21099`)
+
+## Delta Lake connector
+
+* Support automatic type coercion for timestamp type columns during in table
+  creation. ({issue}`19336`)
+* Ensure deletion of blobs ending in slash when deleting directory contents and
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-443.md:13]: @electrum and @martint .. do we need to consider have a separate section for all native file systems .. or one for each file system ..  we could make all the release notes headers to be links to the relevant docs to help with that

File: docs/src/main/sphinx/release/release-443.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Release 443 (21 Mar 2024)
+
+## General
+
+* Ensure consistent formatting when casting from double or real to varchar. ({issue}`21136`)
+* Prevent query failure when partial aggregation over long decimals is pushed
```
Comment (martint) [inline docs/src/main/sphinx/release/release-443.md:6]: "Long decimal" is an implementation detail. Replace with "decimals with precision larger than 18"

File: docs/src/main/sphinx/release/release-443.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Release 443 (21 Mar 2024)
+
+## General
+
+* Ensure consistent formatting when casting from double or real to varchar. ({issue}`21136`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-443.md:5]: Fix formatting of casts from double or real to varchar when running with non-US locales.

File: docs/src/main/sphinx/release/release-443.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+* Prevent query failure when partial aggregation over long decimals is pushed
+  below joins. ({issue}`21099`)
+
+## Delta Lake connector
+
+* Support automatic type coercion for timestamp type columns during in table
```
Comment (martint) [inline docs/src/main/sphinx/release/release-443.md:11]: This is not correct. It's not about automatic type coercion. It's about allowing the creating of tables when the type is specified as something other than timestamp(3). Trino will substitute for a timestamp(3) in that case. We did this for Iceberg a while ago, so we should dig up the release note from that.

File: docs/src/main/sphinx/release/release-443.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  creation. ({issue}`19336`)
+* Ensure deletion of blobs ending in slash when deleting directory contents and
+  using the native S3 file system support. ({issue}`21111`)
+
+## Hive connector
+
+* Reduce coordinator CPU and memory usage. ({issue}`21075`)
+* Prevent failures when listing columms of a table that is concurrently dropped
+  and `sql-standard` authorization is used. ({issue}`21109`)
+* Ensure deletion of blobs ending in slash when deleting directory contents and
```
Comment (martint) [inline docs/src/main/sphinx/release/release-443.md:21]: This seems like an implementation detail. Under what scenarios would those blobs not have been deleted? 

I would rephrase as: "Ensure all files are deleted when <user triggered action> and native S3 filesystem support is enabled"

**Review by martint — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-443.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-443.md:21]: There was another bug about paths with and without ending in slash being treated differently.

File: docs/src/main/sphinx/release/release-443.md (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-443.md:11]: I looked back all the way to 415 .. could not find anything related

File: docs/src/main/sphinx/release/release-443.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+* Fix formatting of casts from double or real to varchar when running with
+  non-US locales. ({issue}`21136`)
+* Prevent query failure when partial aggregation over decimals with precision
+  larger than 18 below joins. ({issue}`21099`)
+
+## Delta Lake connector
+
+* Automatically use `timestamp(3)` as type during table creation when other timestamp precision is specified. ({issue}`19336`)
+* Ensure all files are deleted when and native S3 file system support is
```
Comment (martint) [inline docs/src/main/sphinx/release/release-443.md:13]: Broken sentence: "are deleted *when and* native S3"

File: docs/src/main/sphinx/release/release-443.md (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-443.md]: But I reworded .. maybe the new one is better

File: docs/src/main/sphinx/release/release-443.md (hunk 6 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-443.md]: Updated all four

**Review by martint — APPROVED** [review]
